api/jobs/gears.py

```diff
@@ -119,12 +132,17 @@ def insert_gear(doc):
     if doc.get("invocation-schema"):
         del(doc["invocation-schema"])
 
+    now = datetime.datetime.utcnow()
+
+    doc['created']  = now
+    doc['modified'] = now
+
     config.db.gears.insert(doc)
```

Comment:
You could set `doc['_id'] = str(bson.ObjectId()) before insertion so you don't have to worry about casting it as a string elsewhere.